### PR TITLE
PROTOTYPE:  Melanie/add 2122 form

### PR DESCRIPTION
**Pull request by melaniegershman**
***do not approve, this is just a prototype***
Here's my first pass at creating form 21-22. Concerns and blockers so far have been documented [here](https://docs.google.com/document/d/1wGggipkb-f2fiJBHtA33P2_KAa3parggGUy53o31-RE/edit#).

Please provide any feedback on the schema and the form-config, and @carolynwill , let me know if any changes need to be made to the form layout.

Thanks!

Screenshots are included below.
screen shot 2018-04-10 at 10 07 55 am
screen shot 2018-04-10 at 10 07 28 am
screen shot 2018-04-10 at 10 07 17 am
screen shot 2018-04-10 at 10 07 09 am
screen shot 2018-04-10 at 10 06 59 am
screen shot 2018-04-10 at 10 06 49 am
screen shot 2018-04-10 at 10 06 37 am
screen shot 2018-04-10 at 10 06 22 am

### CODE REVIEW

File: src/js/veteran-representative/2122-schema.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "vaFileNumber": {
+      "type": "string"
+    },
+    "insuranceNumber": {
+      "type": "string"
+    },
+    "claimantFullName": {
+      "type": "string"
+    },
+    "claimantAddress": {
```
Comment (jbalboni) [inline src/js/veteran-representative/2122-schema.json]: We have a common address schema that separates out the fields, so we should switch to using that. It's in vets-json-schema, for reference.

File: src/js/veteran-representative/2122-schema.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      "type": "string"
+    },
+    "claimantEmail": {
```
Comment (jbalboni) [inline src/js/veteran-representative/2122-schema.json]: This should also have a `format: 'email'` property, so that we take advantage of the default JSON Schema email validation logic.

File: src/js/veteran-representative/2122-schema.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        },
+        "suffix": {
+          "type": "string"
+        }
+      }
+    }
+  },
+  "properties": {
+    "veteranFullName": {
+      "type": "string"
```
Comment (jbalboni) [inline src/js/veteran-representative/2122-schema.json]: If you change this line to `"$ref": "#/definitions/fullName"` you'll pull in the common fullName definition from above.

File: src/js/veteran-representative/config/form.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+          schema: {
+            type: "object",
+            required: [
+              'fullName',
+              'veteranSSN',
+              'vaFileNumber',
+              'insuranceNumber',
+            ],
+            properties: {
+              fullName,
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: I think you want this to be `veteranFullName`, since `fullName` is the name of the common definition but not the actual field in the schema.

File: src/js/veteran-representative/config/form.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+    veteranInformation: {
+      title: "Veteran Information",
+      pages: {
+        veteranInformation: {
+          path: "veteran-information",
+          title: "Veteran information",
+          uiSchema: {
+            fullName: fullNameUI,
+            veteranSSN: ssnUI,
+            vaFileNumber: { "ui:title": "VA file number" },
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: I think we have a common definition for this, too.

File: src/js/veteran-representative/config/form.js [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+                "ui:title": "Claimant’s last name",
+              },
+              middle: {
+                "ui:title": "Claimant’s middle name",
+              },
+              suffix: {
+                "ui:title": "Claimant’s suffix",
+              },
+            }),
+            // FIX: need a default field for dropdown
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: As long as the schema for this field has an `enum` property, it should turn into a drop down automatically.

File: src/js/veteran-representative/config/form.js [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+              organizationRepresentativeName,
+              organizationRepresentativeTitle,
+            },
+          },
+        },
+      },
+    },
+    authorizationForRepresentativeAccessToRecords: {
+      title:
+        "Authorization for Representative's Access to Records Protected by Section 7332, TITLE 38, U.S.C.",
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: This is a very long title. More of a design issue, though.

File: src/js/veteran-representative/config/form.js [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+          },
+        },
+      },
+    },
+    claimantInformation: {
+      title: "Claimant Information",
+      pages: {
+        claimantInformation: {
+          path: "claimant-information",
+          title: "Claimant Information",
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: Typically we use sentence case for page titles and title case for chapter titles.

**Review by jbalboni — COMMENTED** [review]
This looks great so far. The majority of the comments I had are just around re-using common definitions. There's also a bunch of code copied over from VIC that can be removed, but I left that alone.

File: src/js/veteran-representative/config/form.js (hunk 7 above)
Comment (melaniegershman) [reply src/js/veteran-representative/config/form.js]: @carolynwill what do you suggest I shorten this to?

File: src/js/veteran-representative/config/form.js (hunk 8 above)
Comment (melaniegershman) [reply src/js/veteran-representative/config/form.js]: Thanks for this feedback, I'll go back and update.

File: src/js/veteran-representative/config/form.js (hunk 6 above)
Comment (melaniegershman) [reply src/js/veteran-representative/config/form.js]: Ahh ok, thank you! If it goes against the grain I won't add it, I wasn't sure what the standard was here.

File: src/js/veteran-representative/config/form.js (hunk 7 above)
Comment (carolynwill) [reply src/js/veteran-representative/config/form.js]: Ooh yeah this one is a doozy. How about we cut everything from "protected...." So it reads "Authorization for Representative's Access to Records"

File: src/js/veteran-representative/2122-schema.json (hunk 1 above)
Comment (melaniegershman) [reply src/js/veteran-representative/2122-schema.json]: Ah! I was on a roll but meant to return to this and use that-- thank you for the reminder!

File: src/js/veteran-representative/definitions/address.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
+import _ from 'lodash/fp';
```
Comment (jbalboni) [inline src/js/veteran-representative/definitions/address.js]: I think this file can be deleted if we can move to the common address definition.

File: src/js/veteran-representative/helpers.jsx [hunk 10]
```diff
@@ -0,0 +1,215 @@
+import _ from 'lodash/fp';
```
Comment (jbalboni) [inline src/js/veteran-representative/helpers.jsx]: I think this file can be deleted.

File: src/js/veteran-representative/config/form.js [hunk 11]
```diff
@@ -0,0 +1,282 @@
+import fullSchema from '../2122-schema.json';
+import _ from 'lodash/fp';
+import IntroductionPage from '../containers/IntroductionPage';
+import ConfirmationPage from '../containers/ConfirmationPage';
+
+import fullNameUI from '../../common/schemaform/definitions/fullName';
+import ssnUI from '../../common/schemaform/definitions/ssn';
+import * as addressUI from '../definitions/address';
```
Comment (jbalboni) [inline src/js/veteran-representative/config/form.js]: This should probably use the address definition from `src/js/common/schemaform/definitions`

File: src/sass/veteran-representative.scss [hunk 12]
```diff
@@ -0,0 +1,8 @@
+@import "shared-variables";
```
Comment (jbalboni) [inline src/sass/veteran-representative.scss]: You'll have to update this file to use the paths that match the other forms, which have changed recently (e.g. https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/burials/sass/burials.scss)

**Review by jbalboni — APPROVED** [review]
